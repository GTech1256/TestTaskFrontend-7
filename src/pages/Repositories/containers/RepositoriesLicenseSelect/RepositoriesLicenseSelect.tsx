import React, { useMemo } from 'react';

import { ALL_LICENSIES, getAllLicensiesQueryType } from '../../../../services/graphql/queries/getAllLicensies';
import Select, { SelectProps } from '../../../../components/Select';
import { useQuery } from '@apollo/client';


type Props = Pick<SelectProps, 'className' | 'name' | 'onChange' | 'value'>

const RepositoriesLicenseSelect = (props: Props) => {
  const { loading, error, data } = useQuery<getAllLicensiesQueryType>(ALL_LICENSIES);

  const options = useMemo<SelectProps['options']>(
    () => !data ? 
      [] :
      data?.licenses.map(license => ({ label: license.name, value: license.key })),
    [data]
    )

  if (error) return <p>Не удалось получить лицении</p>

  return <Select
    title="Лицензия"
    {...props}
    isLoading={loading}
    options={options}
  />
}

export default RepositoriesLicenseSelect;