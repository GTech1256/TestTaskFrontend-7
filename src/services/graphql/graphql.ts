import { setContext } from '@apollo/client/link/context';
import { ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';

import { getGithubToken } from '../github';


const httpLink = createHttpLink({
  uri: 'https://api.github.com/graphql',
});

const authLink = setContext(async (_, { headers }) => {
  const token = await getGithubToken();

  // return the headers to the context so httpLink can read them
  return {
    headers: {
      ...headers,
      authorization: token ? `Bearer ${token}` : "",
    }
  }
});

export const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache()
});