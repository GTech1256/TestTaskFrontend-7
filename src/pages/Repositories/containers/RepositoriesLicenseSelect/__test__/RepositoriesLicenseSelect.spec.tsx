import React from 'react';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';

import RepositoriesLicenseSelect from '../RepositoriesLicenseSelect';
import { client } from '../../../../../services/graphql';


describe('<RepositoriesLicenseSelect />', () => {
  it('should render correct', () => {
    const props = {
      name: 'test',
      onChange: () => {},
      value: undefined
    };

    const tree = renderer
      .create(<ApolloProvider client={client}><RepositoriesLicenseSelect {...props} /></ApolloProvider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
})