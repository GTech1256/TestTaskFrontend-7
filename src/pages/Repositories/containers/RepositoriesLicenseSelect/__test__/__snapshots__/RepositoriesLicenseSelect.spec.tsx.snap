// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesLicenseSelect /> should render correct 1`] = `
<div
  className="input-group mb-3 "
>
  <div
    className="input-group-prepend"
  >
    <label
      className="input-group-text"
      htmlFor="test"
    >
      Лицензия
    </label>
  </div>
  <select
    className="custom-select"
    defaultValue="DEFAULT"
    disabled={true}
    id="test"
    onChange={[Function]}
  >
    <option
      value="DEFAULT"
    >
      Фильтр по лицензии...
    </option>
  </select>
</div>
`;
